bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/veneer/BigtableHBaseVeneerSettings.java

```diff
@@ -0,0 +1,580 @@
+      stubSettings.setCredentialsProvider(FixedCredentialsProvider.create(credentials));
+    }
+  }
+
+  // copied over from CredentialFactory
+  // TODO: Find a better way to convert P12 key into Credentials instance
+  private Credentials getCredentialFromPrivateKeyServiceAccount(
+      String serviceAccountEmail, String privateKeyFile) throws IOException {
+    try {
+      PrivateKey privateKey =
+          SecurityUtils.loadPrivateKeyFromKeyStore(
+              SecurityUtils.getPkcs12KeyStore(),
+              new FileInputStream(privateKeyFile),
+              "notasecret",
+              "privatekey",
+              "notasecret");
+
+      return ServiceAccountJwtAccessCredentials.newBuilder()
+          .setClientEmail(serviceAccountEmail)
+          .setPrivateKey(privateKey)
+          .build();
+    } catch (GeneralSecurityException exception) {
+      throw new RuntimeException("exception while retrieving credentials", exception);
+    }
+  }
+
+  /** Creates {@link TransportChannelProvider} for plaintext negotiation type. */
+  private TransportChannelProvider buildPlainTextChannelProvider(String endpoint) {
+
+    InstantiatingGrpcChannelProvider.Builder channelBuilder =
+        defaultGrpcTransportProviderBuilder()
+            .setEndpoint(endpoint)
+            .setChannelConfigurator(
+                new ApiFunction<ManagedChannelBuilder, ManagedChannelBuilder>() {
+                  @Override
+                  public ManagedChannelBuilder apply(ManagedChannelBuilder channelBuilder) {
+                    return channelBuilder.usePlaintext();
+                  }
+                });
+
+    String channelCount = configuration.get(BIGTABLE_DATA_CHANNEL_COUNT_KEY);
+    if (!isNullOrEmpty(channelCount)) {
+      channelBuilder.setPoolSize(Integer.parseInt(channelCount));
+    }
+
+    return channelBuilder.build();
+  }
+
+  private Set<StatusCode.Code> extractRetryCodesFromConfig() {
+    ImmutableSet.Builder<StatusCode.Code> statusCodeBuilder = ImmutableSet.builder();
+    String retryCodes = configuration.get(ADDITIONAL_RETRY_CODES, "");
+
+    for (String stringCode : retryCodes.split(",")) {
+      String trimmed = stringCode.trim();
+      if (trimmed.isEmpty()) {
+        continue;
+      }
+
+      StatusCode.Code code = StatusCode.Code.valueOf(trimmed);
+      Preconditions.checkNotNull(code, String.format("Unknown status code %s found", stringCode));
+      statusCodeBuilder.add(code);
+      LOG.debug("gRPC retry on: %s", stringCode);
+    }
+    return statusCodeBuilder.build();
+  }
+
+  private <Request, Response> void buildIdempotentCallSettings(
+      UnaryCallSettings.Builder<Request, Response> unaryCallSettings) {
+
+    ImmutableSet.Builder<StatusCode.Code> retryCodeBuilder = ImmutableSet.builder();
+    if (configuration.getBoolean(ENABLE_GRPC_RETRIES_KEY, true)) {
+      retryCodeBuilder
+          .addAll(extractRetryCodesFromConfig())
+          .addAll(unaryCallSettings.getRetryableCodes());
+    }
```

Comment:
shouldn't this have an else?
ie if config.set(ENABLE_GRPC_RETRIES_KEY, false). Then shouldn't we disable retries? ie clear retry codes